src/pk-transaction-db.c

```diff
@@ -288,83 +288,207 @@ pk_transaction_db_get_list (PkTransactionDb *tdb, guint limit)
+
+	if (rc == SQLITE_OK || rc == SQLITE_DONE) {
+		rc = SQLITE_OK;
+	} else {
+		g_warning ("(%s) SQL error: %d: %s", sql, rc, sqlite3_errmsg (db));
+	}
+
+	return rc;
+}
+
+static gboolean
+pk_transaction_db_set_strings2 (PkTransactionDb *tdb, const gchar *sql, const gchar *first, const gchar *second)
+{
+	sqlite3_stmt *statement = NULL;
+	gint rc = 0;
 
 	g_return_val_if_fail (PK_IS_TRANSACTION_DB (tdb), FALSE);
+	g_return_val_if_fail (tdb->priv->db != NULL, FALSE);
+	g_return_val_if_fail (sql != NULL, FALSE);
+	g_return_val_if_fail (first != NULL, FALSE);
+	g_return_val_if_fail (second != NULL, FALSE);
 
-	role_text = pk_role_enum_to_string (role);
-	statement = g_strdup_printf ("UPDATE transactions SET role = '%s' WHERE transaction_id = '%s'", role_text, tid);
-	pk_transaction_db_sql_statement (tdb, statement);
-	return TRUE;
+	if (!(statement = pk_transaction_db_prepare (tdb, sql))) {
+		goto out;
+	}
+
+	if ((rc = sqlite3_bind_text (statement, 1, first, -1, SQLITE_STATIC)) != SQLITE_OK) {
+		g_warning ("(%s) bind text1 error: %d: %s", sql, rc, sqlite3_errmsg (tdb->priv->db));
+		goto out;
+	}
+
+	if ((rc = sqlite3_bind_text (statement, 2, second, -1, SQLITE_STATIC)) != SQLITE_OK) {
+		g_warning ("(%s) bind text2 error: %d: %s", sql, rc, sqlite3_errmsg (tdb->priv->db));
+		goto out;
+	}
+
+	rc = pk_transaction_db_step (tdb->priv->db, statement, sql);
+
+out:
+	sqlite3_finalize (statement);
+
+	return rc == SQLITE_OK;
 }
 
-gboolean
-pk_transaction_db_set_uid (PkTransactionDb *tdb, const gchar *tid, guint uid)
+static gboolean
+pk_transaction_db_set_ints1_strings1 (PkTransactionDb *tdb, const gchar *sql, gint first, const gchar *second)
 {
-	g_autofree gchar *statement = NULL;
+	sqlite3_stmt *statement = NULL;
+	gint rc = 0;
 
 	g_return_val_if_fail (PK_IS_TRANSACTION_DB (tdb), FALSE);
+	g_return_val_if_fail (tdb->priv->db != NULL, FALSE);
+	g_return_val_if_fail (sql != NULL, FALSE);
+	g_return_val_if_fail (second != NULL, FALSE);
 
-	statement = g_strdup_printf ("UPDATE transactions SET uid = '%i' WHERE transaction_id = '%s'", uid, tid);
-	pk_transaction_db_sql_statement (tdb, statement);
-	return TRUE;
+	if (!(statement = pk_transaction_db_prepare (tdb, sql))) {
+		goto out;
+	}
+
+	if ((rc = sqlite3_bind_int (statement, 1, first)) != SQLITE_OK) {
+		g_warning ("(%s) bind int error: %d: %s", sql, rc, sqlite3_errmsg (tdb->priv->db));
+		goto out;
+	}
+
+	if ((rc = sqlite3_bind_text (statement, 2, second, -1, SQLITE_STATIC)) != SQLITE_OK) {
+		g_warning ("(%s) bind text error: %d: %s", sql, rc, sqlite3_errmsg (tdb->priv->db));
+		goto out;
+	}
+
+	rc = pk_transaction_db_step (tdb->priv->db, statement, sql);
+
+out:
+	sqlite3_finalize (statement);
+
+	return rc == SQLITE_OK;
 }
 
-gboolean
-pk_transaction_db_set_cmdline (PkTransactionDb *tdb, const gchar *tid, const gchar *cmdline)
+static gboolean
+pk_transaction_db_set_ints2_strings1 (PkTransactionDb *tdb, const gchar *sql, gint first, gint second, const gchar *third)
```

Comment:
I really don't like this template style, sorry; I think sqlite3_bind_int just has to go in the caller.

Reply:
That's fine, I just went a bit overboard with the functions. Other than those where two strings are applied these are not used elsewhere so no point in having separate functions.